modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/controller/PortletExportControllerImpl.java

```diff
@@ -195,7 +195,10 @@ public void exportAssetLinks(PortletDataContext portletDataContext)
 
 			List<AssetLink> assetLinks = new ArrayList<>();
 
-			if (_isIncludeAllAssetLinks()) {
+			if (_isIncludeAllAssetLinks() &&
```

Comment:
Hi @antonio-ortega ,

**_isIncludeAllAssetLinks** returns false by the default configs, so I'm not sure if the new code runs at all.

Is there something wrong with the OSGI config?
Or should we add extra steps to the LPS description?

Best,
Tamas

Reply:
Hi @moltam89 ,

Thanks for fast reply.

Yes, you need an extra step in Master enabling such property. 

Just updated LPS accordingly.

Thanks!

Reply:
Uhm, I'm still confused a little bit, the OSGI config was created to solve this performance problem.

Is there a customer ticket for this on an old branch, where the config doesn't exist? Can't we just backport the config?

If we apply the new code, the config won't work anymore, I assume the new code will decide about the behaviour.

Reply:
Yes, this was reported in 7.3.x where this config is not available and actually I don't think backporting this to a maintenance branch is the right solution since we would be changing default behavior and it could be considered as a breaking change.

Maybe we can say this issue is not affecting Master, since the user needs to consciously enable the checkbox "Include all asset links" so they know they might be running into trouble if they have a large amount of Asset Links.

Even if we consider that option, considering this case only for maintenance branches, I might say, from my point of view, this OSGI config might luckily fix the issue but it could be confusing for an end user.

Have a think to the case when you want to include some Asset Links, to be more exact, those AssetLinks create in the date range you selected. How can we achieve that goal?

- If we enable "Include all asset links" we might think we are including all of them, not only those within the Date Range.
- If we don't enable "Include all asset links" we might think we are not including all, so we might be including only those within the Date Range... but we are not including anything based on the Date Range.

Just my two cents.

As I said above, if you are happy with it, you can close this pull request and consider this an issue only until 7.3.x.

Thanks for your quick feedback!

Reply:
Okay, I see, [LPS-83011](https://issues.liferay.com/browse/LPS-83011) was a Story ticket, I think it would be safe to backport it though. For older branches we could set the default value to true, so it won't change the behavior by default, but customers who encounter this issue could change the config.

In master, we pass in portletDataContext.getStartDate() and portletDataContext.getEndDate().
Is there maybe a bug there, how these dates are handled?
On the steps from the LPS, the "All" radio button is selected, in case of a date range, I'd expect to only export the asset links which fits to the date range.

Maybe the config's description is misleading a little bit, "Include All Asset Links"  is not necessarily all the links.
I think the date range is still considered, by "All" originally I meant that not only those asset links, which are connected to the exported content.

Thanks

Reply:
Just a quick note, when you say:

> _On the steps from the LPS, the "All" radio button is selected, in case of a date range, I'd expect to only export the asset links which fits to the date range._

The "All" radio button don't apply in case of a date range. It's one or the other.

<img width="772" alt="image" src="https://user-images.githubusercontent.com/14979768/169506078-8c7611b1-4ffc-4544-a925-5f8b5cd3a7aa.png">

The issue reported by our customer is happening in 7.3.x when "All" radio button is selected because they are exporting all asset links (they have above a million) and they don't know that's happening. They think they are just exporting a few web contents and docs because that's what you can see in the summary below.

They noticed the issue was not happening when selected a date range, what initially was a complete mistery until we realized in that case they were exporting just a few Asset Links.

I agree with the idea of backporting OSGI config and set "true" as default value so we don't break anything, just in case. So feel free to close this pull request, I'll send a new one to 7.3.x branch backporting LPS-83011 and changing default value.

I also agree, we might slightly correct the description for that property, basically I can see 3 use cases in Master as of today:

1. Out of the box, you don't export any asset link.
2. If you enable that property, then you export all asset links.
3. If you enable that property and select data by date range, then you export all asset links within the date range.

Current description is covering cases 1 and 2 but not 3.

Reply:
I sent a PR to fix the description.
https://github.com/liferay-staging/liferay-portal/pull/63

I'm closing this PR.

Thanks,
Tamas